### PR TITLE
fix: source bridges missing after restore the backup files

### PR DESCRIPTION
**Pull request by terry-xiaoyu**
The following configs in backup files cannot be imported:

```
sources {
  mqtt {
    source_c384b174 {
      connector = source_connector_c8287217
      enable = true
      parameters {
        qos = 0
        topic = "CHESAPEAKE/Writeback/+/+"
      }
    }
  }
}
```

Release version: v5.6.1

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx_bridge/src/emqx_bridge_v2.erl [hunk 1]
```diff
@@ -1030,7 +1030,26 @@ bridge_v2_type_to_connector_type(Type) ->
...
+    Results = lists:foldr(
+        fun
+            ({ok, OkRes}, {OkAcc, ErrAcc}) ->
+                {[OkRes | OkAcc], ErrAcc};
+            ({error, ErrRes}, {OkAcc, ErrAcc}) ->
+                {OkAcc, [ErrRes | ErrAcc]}
+        end,
+        {[], []},
+        Results0
+    ),
```
Comment (thalesmg) [inline apps/emqx_bridge/src/emqx_bridge_v2.erl:1051]: Minor suggestion:

```suggestion
    Results = lists:partition(fun({ok, _}) -> true; ({error, _}) -> false end, Results0),
```

File: changes/ce/fix-12826.en.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+Cannot import sources actions and retained messages.
+
+Before the fix:
+
+- source actions are ignored from the backup file
```
Comment (thalesmg) [inline changes/ce/fix-12826.en.md:5]:
```suggestion
- source data integrations are ignored from the backup file
```

File: changes/ce/fix-12826.en.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+Cannot import sources actions and retained messages.
```
Comment (thalesmg) [inline changes/ce/fix-12826.en.md:1]:
```suggestion
Fixed an issue that prevented importing source data integrations and retained messages.
```

**Review by thalesmg — APPROVED** [review]

File: apps/emqx_retainer/src/emqx_retainer_mnesia.erl [hunk 4]
```diff
@@ -73,6 +76,12 @@
 topics() ->
     [emqx_topic:join(I) || I <- mnesia:dirty_all_keys(?TAB_MESSAGE)].
 
+%%--------------------------------------------------------------------
+%% Data backup
+%%--------------------------------------------------------------------
+backup_tables() ->
+    [?TAB_MESSAGE].
```
Comment (SergeTupchiy) [inline apps/emqx_retainer/src/emqx_retainer_mnesia.erl:83]: Note: backup/restore won't work for `ram_copies` table which is a default storage type for `?TAB__MESSAGE`...
It doesn't seem to fail, but the content of the backup is empty.

Comment (SergeTupchiy) [reply apps/emqx_retainer/src/emqx_retainer_mnesia.erl:83]: Looks like ram_copies can be backed up by adding `{ram_overrides_dump,true}]` opt:
```
mnesia:activate_checkpoint([{name, emqx_retainer_message}, {min, [emqx_retainer_message]}, {allow_remote, false}, {ram_overrides_dump,true}]).
```

Comment (terry-xiaoyu) [reply apps/emqx_retainer/src/emqx_retainer_mnesia.erl:83]: Thanks a lot for the explanation, I didn't know this before!

I thinks it won't cause problems as now the backup/restore of the retained messages are handled as following:

1. we required the user to backup the retained messages by coping the `emqx_retainer.DCD` and `emqx_retainer.DCL` dump files manually. If it is a `ram_copies` table, then one has to do `emqx eval 'mnesia:dump_tables([emqx_retainer])'` to generate a dump file.
2. convert the retained messages table using our [converter](https://github.com/emqx/emqx-data-converter), and provide a --data-files-dir option to the dir that contains dump files. See this PR for more information: https://github.com/emqx/emqx-data-converter/pull/6
3. import the generated tarball to emqx 5.6.x

File: apps/emqx_bridge/src/emqx_bridge_v2.erl (hunk 1 above)
Comment (terry-xiaoyu) [reply apps/emqx_bridge/src/emqx_bridge_v2.erl:1051]: But I also need to remove the `ok` and `error` tags from the result list, so `lists:partition/2` is not appropriate.

**Review by HJianBo — APPROVED** [review]

File: apps/emqx_retainer/src/emqx_retainer_mnesia.erl (hunk 4 above)
Comment (HJianBo) [reply apps/emqx_retainer/src/emqx_retainer_mnesia.erl:83]: @terry-xiaoyu will open another PR to fix it.

Comment (SergeTupchiy) [reply apps/emqx_retainer/src/emqx_retainer_mnesia.erl:83]: Hi @HJianBo, @terry-xiaoyu,
Just FYI, I've fixed backup export for RAM storage tables: https://github.com/emqx/emqx/pull/13293/commits/baa79962891ff3a597d8d9dfdf465faf46e095d6
It's now possible to export retained messages even if they are stored in ram.

**Pull request merged**